nixos/modules/services/misc/zoneminder.nix

```
@@ -204,10 +206,9 @@ in {
       };
 
       mysql = lib.mkIf cfg.database.createLocally {
+        enable = true;
+        package = lib.mkDefault pkgs.mariadb;
         ensureDatabases = [ cfg.database.name ];
-        initialDatabases = [{
-          inherit (cfg.database) name; schema = "${pkg}/share/zoneminder/db/zm_create.sql";
-        }];
         ensureUsers = [{
           name = cfg.database.username;
```

```suggestion
           name = user;
```

Because when used locally the Linux user is used for socket authentication as seen in this line:

```
     ZM_DB_USER=${if cfg.database.createLocally then user else cfg.database.username}
```

---

@JohnAZoidberg I have taken care of this with an assertion instead.

---

Then you have to revert the comment `Not used if <literal>createLocally</literal> is set.` for the database username.
And I'd also default it to the Linux user if `createLocally` ist set, for conenience.